everestrs/everestrs-build/src/codegen.rs

```diff
@@ -307,12 +315,145 @@ impl CommandContext {
     }
 }
 
+/// The error group maps to one error yaml file.
+#[derive(Debug, Clone, Serialize)]
+struct ErrorGroupContext {
+    /// The name is basically the yaml file in which the errors are defined.
+    name: String,
+
+    /// The list of errors
+    error_list: schema::error::ErrorList,
+}
+
+mod impl_error {
+    #[derive(Hash, Eq, PartialEq)]
+    pub struct ErrorPath<'a> {
+        /// The prefix where the error files are.
+        pub prefix: &'a str,
+
+        /// The error file itself.
+        pub file: &'a str,
+    }
+
+    pub struct ErrorDefinition<'a> {
+        /// The path of the error.
+        pub path: ErrorPath<'a>,
+
+        /// The type which is optional. If the type is not defined we accept
+        /// all errors in the path.
+        pub error_type: Option<&'a str>,
+    }
+
+    impl<'a> ErrorDefinition<'a> {
+        /// Try to construct an error definition from the string.
+        pub fn try_new(value: &'a str) -> anyhow::Result<Self> {
+            let mut splits = value.split("#/");
+            let path = splits.next().ok_or(anyhow::anyhow!("No path defined"))?;
+
+            // Split the path and remove the empty parts.
+            // (The first element might be empty if we have a leading `/`).
+            let paths = path
+                .split("/")
+                .filter(|path| !path.is_empty())
+                .collect::<Vec<_>>();
+
+            anyhow::ensure!(paths.len() == 2);
```

Comment:
Is the message printed if this is not fulfilled meaningful enough? I believe it will just print `Error: Condition failed: `paths.len() == 2` (1 vs 2)`. Same happens with the line below this one